src/main/scala/org/scalatestplus/junit/JUnitRunner.scala

```diff
@@ -113,5 +114,10 @@ final class JUnitRunner(suiteClass: java.lang.Class[_ <: Suite]) extends org.jun
    *  @return the expected number of tests that will run when this suite is run
    */
   override def testCount() = suiteToRun.expectedTestCount(Filter())
+
+  override def filter(filter: TestFilter): Unit = {
+    if (filter.shouldRun(getDescription)) throw new NoTestsRemainException
```

Comment:
I think you need to negate the condition. Towards that end, is a test possible?

Reply:
You are right, my bad. I added a test to check the correct behavior